src/services/users.service.ts

```diff
@@ -52,6 +52,17 @@ class UserService {
     const deleteUserData = await this.users.delete({ where: { id: userId }, select: { email: true, id: true } });
     return deleteUserData;
   }
+
+  public async manageGroups(userId: number, data: any) {
+    if (isEmpty(data)) throw new HttpException(400, "Groups don't exist");
+    
+    const groups = await this.users.update({
+      where: { id: userId },
+      data
+    })
+
+    return groups;
+  }
```

Comment:
This function has some things that look of:
1. Line 57: The message is telling us that a resource doesn't exist but, actually we are throwing the exception because of the validation wasn't fulfilled, so I would change the message.
2. The function is named `manageGroups`, but in fact, is updating user's records.
3. Is risky that parameter `data` is `any`; I would type that to ensure we are passing compliant data to Prisma.